components/admin_console/feature_discovery/feature_discovery.tsx

```diff
@@ -75,9 +78,9 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
             users = this.props.stats.TOTAL_USERS;
         }
         const requestedUsers = Math.max(users, 30);
-        const {error} = await this.props.actions.requestTrialLicense(requestedUsers, true, true, this.props.featureName);
+        const {error, data} = await this.props.actions.requestTrialLicense(requestedUsers, true, true, this.props.featureName);
         if (error) {
-            this.setState({gettingTrialError: error});
+            this.setState({gettingTrialError: error, gettingTrialResponseCode: data.status});
```

Comment:
In case of any other error different to 451 `data` will be undefined, and `data.status` would throw a type error.

Reply:
fixed

Reply:
I think this still needs a tweak, something along the lines of

```ts
if (typeof data?.status === 'undefined') {
```

data itself can be undefined

Reply:
Woops, good catch